compiler/rustc_mir_transform/src/shim.rs

```diff
@@ -1005,6 +1013,609 @@ fn build_fn_ptr_addr_shim<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId, self_ty: Ty<'t
     new_body(source, IndexVec::from_elem_n(start_block, 1), locals, sig.inputs().len(), span)
 }
 
+fn build_async_destructor_ctor_shim<'tcx>(
```

Comment:
move this new code into a separate module. It makes up 1/3rd of this module 😆

Reply:
Done in c2eff1fb0510cf371a5f483b8a0e9af572132369 🙂